README.md

```diff
@@ -1,15 +1,188 @@
+
+## parse_signature
+
+This macro generates a SQL that transforms each raw event into it’s signature.  
+Instead of using the signature from the column definitions, it is deriving the signature from the value. This can be used to compare events with identical names together to understand if and how they vary from each other.
+
+
+**Raw Events:**
+| event_name  | user_id      | user_name | device  |
+|-------------|--------------|-----------|---------|
+| App Opened  | 2c3416a3033e | John      | IOS     |
+| App Opened  | fc3416basdq1 | Donna     | IOS     |
+| App Opened  | c5e6d9429f27 | Lisa      |         |
+| App Opened  | 55331212412  | Joshua    | Android |
+
+
+
+**After running parse_signature:**
+| event_name | property_name_mapping        | property_signature_mapping |
+|------------|------------------------------|----------------------------|
+| App Opened | [user_id, user_name, device] | [string, string, string]   |
+| App Opened | [user_id, user_name, device] | [string, string, string]   |
+| App Opened | [user_id, user_name, device] | [string, string, null]     |
+| App Opened | [user_id, user_name, device] | [int, string, null]        |
+
+
+We recommend to test this in your development environment first while you get set up.
+```
+{# in dbt Develop #}
+
+{% set raw_event_relation=adapter.get_relation(
+      database=target.database,
+      schema="raw_event_schema",
+      identifier="raw_event_table"
+) -%}
+
+{{ avo_audit.parse_signature(
+    relation=raw_event_relation,
+    date_partition=”sent_at”,
+    lookback_hours=1,
+) }}
+```
+
+**Arguments:**
+* **raw_event_relation:** The database table containing your raw events which you want to compare.
+* **date_partition:** The column in raw events that can be used to filter on time to limit query.
+* **lookback_hours:** How many hours the query should look back, we recommend keeping it under 24h to keep the cost of the query low.
+
+
+
+## compare_events
+
+This macro will run on the results of parse_signature.
+
+It will execute a SQL that will go through each event_name group, and compare it together with all the other events to identify the issues documented above.
+
+The end result will be a table that identifies discrepancies between the events.
+
+| event_name | property_name_mapping        | property_signature_mapping | variant_count | total_event_count | number_of_event_variants |
+|------------|------------------------------|----------------------------|---------------|-------------------|--------------------------|
+| App Opened | [user_id, user_name, device] | [string, string, string]   | 2             | 4                 | 3                        |
+| App Opened | [user_id, user_name, device] | [string, string, null]     | 1             | 4                 | 3                        |
+| App Opened | [user_id, user_name, device] | [int, string, null]        | 1             | 4                 | 3                        |
+
+
```

Comment:
Add some context for the commands here? Like "To execute, run the following commands in your dbt dev environment:"